### PR TITLE
adds ObsPy as a python package

### PR DESCRIPTION
**Pull request by francislz**
FYI Professor @hyviquel

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-aiobotocore/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyAiobotocore(PythonPackage):
     homepage = "https://aiobotocore.readthedocs.io/en/latest/"
     pypi = "aiobotocore/aiobotocore-1.2.1.tar.gz"
 
+    version("2.5.2", sha256="e7399f21570db1c287f1c0c814dd3475dfe1c8166722e2c77ce67f172cbcfa89")
     version("2.5.0", sha256="6a5b397cddd4f81026aa91a14c7dd2650727425740a5af8ba75127ff663faf67")
     version("2.4.2", sha256="0603b74a582dffa7511ce7548d07dc9b10ec87bc5fb657eb0b34f9bd490958bf")
     version("1.2.1", sha256="58cc422e65fc89f7cb78eca740d241ac8e15f39f6b308cc23152711e8a987d45")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aiobotocore/package.py]: Need to add upper bounds on deps

File: var/spack/repos/builtin/packages/py-annotated-types/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    homepage = "https://pypi.org/project/annotated-types"
+    pypi = "annotated-types/annotated_types-0.5.0.tar.gz"
+
+    version("0.5.0", sha256="47cdc3490d9ac1506ce92c7aaa76c579dc3509ff11e098fc867e5130ab7be802")
+
+    depends_on("python@3.9", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-typing-extensions@4.1.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-annotated-types/package.py]:
```suggestion
    depends_on("py-hatchling", type="build")
    depends_on("py-typing-extensions@4:", when="^python@:3.8", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-datetimerange/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    pypi = "DateTimeRange/DateTimeRange-2.1.0.tar.gz"
+
+    maintainers("francislz")
+
+    version("2.1.0", sha256="944e70d964c95618a826e82a8876762bf9acc1028d9d8a7fb425f31d1abe4ef8")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-python-dateutil", type=("build", "run"))
+    depends_on("py-typepy@1.3.1", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-datetimerange/package.py]:
```suggestion
    depends_on("py-python-dateutil@2.4.2:2", type=("build", "run"))
    depends_on("py-typepy@1+datetime", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-mbstrdecoder/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://pypi.org/project/mbstrdecoder"
+    pypi = "mbstrdecoder/mbstrdecoder-1.1.3.tar.gz"
+
+    maintainers("francislz")
+
+    version("1.1.3", sha256="dcfd2c759322eb44fe193a9e0b1b86c5b87f3ec5ea8e1bb43b3e9ae423f1e8fe")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-chardet", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mbstrdecoder/package.py]:
```suggestion
    depends_on("py-setuptools@61:", type="build")
    depends_on("py-chardet@3.0.4:5", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-noisepy-seis/package.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    homepage = "https://github.com/mdenolle/NoisePy"
+    pypi = "noisepy-seis/noisepy_seis-0.9.71.tar.gz"
+
+    maintainers("francislz")
+
+    version("0.9.71", sha256="99cd2a8e4a8b40e206054147994fcb191d4605ac52abcde495390cd67c58288b")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-noisepy-seis/package.py]:
```suggestion
    depends_on("python@3.9:3.10", type=("build", "run"))
    depends_on("py-hatchling", type="build")
    depends_on("py-hatch-vcs", type="build")
```

File: var/spack/repos/builtin/packages/py-noisepy-seis/package.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    depends_on("py-s3fs@2023.4.0:", type="build")
+    depends_on("py-zarr@2.10.2:", type=("build", "run"))
+    depends_on("py-aiobotocore@2.5.2", type=("build", "run"))
+    depends_on("py-pydantic@2.0", type=("build", "run"))
+    depends_on("py-pyyaml@6.0", when="@2.9:", type=("build", "run"))
+    depends_on("py-psutil@5.9.5:", type=("build", "run"))
+    depends_on("ffmpeg+openssl", type="run")
+    depends_on("py-datetimerange@2.0.0:", type=("build", "run"))
+    depends_on("py-pycwt@0.4.0b0:", type=("build", "run"))
+    depends_on("py-pydantic-yaml@1.1.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-noisepy-seis/package.py]: Many of these are missing upper bounds:
```toml
dependencies = [
    "DateTimeRange>=2.0.0,<3.0.0",
    "numpy>=1.22.0, <2.0.0",
    "h5py>=3.8.0,<4.0.0",
    "pandas>=1.5.3,<2.0.0",
    "numba>=0.57.0,<1.0.0",
    "pyasdf>=0.7.5,<1.0.0",
    "pycwt>=0.3.0a22,<1.0.0",
    "diskcache>=5.6.1,<6.0.0",
    "fsspec>=2023.4.0,<2024.0.0",
    "s3fs==2023.4.0,<2024.0.0",
    "zarr>=2.14.2,<3.0.0",
    "aiobotocore==2.5.2", # "2.5.3 is broken"
    "pydantic==2.0.0",
    "PyYAML==6.0",
    "pydantic-yaml==1.0",
]
```

File: var/spack/repos/builtin/packages/py-pycwt/package.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    pypi = "pycwt/pycwt-0.4.0b0.tar.gz"
+
+    version("0.4.0b0", sha256="f337cd28531f5b49a62b5f3ba5657cbc3e2f69e15a12ee3d309a2aa5ca1b8e86")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-numpy@1.22.0:", type=("build", "run"))
+    depends_on("py-scipy@1.7.0:", type=("build", "run"))
+    depends_on("py-matplotlib@3.4.0:", type=("build", "run"))
+    depends_on("py-tqdm@4.62.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pycwt/package.py]: Deps are incorrect:
```toml
requires = ["setuptools>=45", "setuptools_scm[toml]>=6.2"]
dependencies = [
    "numpy>=1.24, <2",
    "scipy>=1.10, <2",
    "matplotlib>=3.7, <4",
    "tqdm>=4.65, <5"
]
requires-python = ">=3.8"
```

File: var/spack/repos/builtin/packages/py-pydantic-core/package.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+class PyPydanticCore(PythonPackage):
+    """This package provides the core functionality for pydantic validation and serialization."""
+
+    homepage = "https://github.com/pydantic/pydantic-core"
+    pypi = "pydantic-core/pydantic_core-2.0.2.tar.gz"
+
+    version("2.0.2", sha256="996ffb7ae3c8cb7506a58dae52bbf13a7bbbfce6c3110a2b44c20d2587e57b9b")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-typing-extensions@4.6.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pydantic-core/package.py]: Deps are incorrect:
```toml
requires = [
    'maturin>=1,<2',
    'typing-extensions >=4.6.0,!=4.7.0'
]
dependencies = [
    'typing-extensions >=4.6.0,!=4.7.0'
]
```

File: var/spack/repos/builtin/packages/py-pydantic-yaml/package.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+    version("1.1.1", sha256="1c7bfcb1c4cb5d74dd3d764b098785297df63d7a4833bd258317a2000b4c5fc7")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", type=("build", "run"))
+
+    depends_on("py-pydantic@1.8.2:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.17.16:", type=("build", "run"))
+    depends_on("py-importlib_metadata@3.6:", type=("build", "run"))
+    depends_on("py-typing_extensions@4.5.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pydantic-yaml/package.py]: Deps are incorrect

File: var/spack/repos/builtin/packages/py-pydantic/package.py [hunk 10]
```diff
@@ -20,6 +21,9 @@ class PyPydantic(PythonPackage):
     variant("dotenv", default=False, description="Install requirements for pydantic.dotenv")
 
     depends_on("py-setuptools", type="build")
+    depends_on("py-pydantic-core@2.0.2:", when="@2.0.1", type=("build", "run"))
+    depends_on("py-typing-extensions@4.6.3:", when="@2.0.1:", type=("build", "run"))
+    depends_on("py-annotated-types@0.5.0:", when="@2.0.1:", type=("build", "run"))
     depends_on("py-typing-extensions@4.2:", when="@1.10.9:", type=("build", "run"))
     depends_on("py-typing-extensions@4.1:", when="@1.10:", type=("build", "run"))
     depends_on("py-typing-extensions@3.7.4.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pydantic/package.py]: Deps are incorrect

File: var/spack/repos/builtin/packages/py-s3fs/package.py [hunk 11]
```diff
@@ -13,6 +13,7 @@ class PyS3fs(PythonPackage):
     homepage = "https://s3fs.readthedocs.io/en/latest/"
     pypi = "s3fs/s3fs-0.5.2.tar.gz"
 
+    version("2023.6.0", sha256="63fd8ddf05eb722de784b7b503196107f2a518061298cf005a8a4715b4d49117")
     version("2022.11.0", sha256="10c5ac283a4f5b67ffad6d1f25ff7ee026142750c5c5dc868746cd904f617c33")
     version("0.5.2", sha256="87e5210415db17b9de18c77bcfc4a301570cc9030ee112b77dc47ab82426bae1")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-s3fs/package.py]: Deps need to be updated

File: var/spack/repos/builtin/packages/py-typepy/package.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://pypi.org/project/typepy"
+    pypi = "typepy/typepy-1.3.1.tar.gz"
+
+    maintainers("francislz")
+
+    version("1.3.1", sha256="dfc37b888d6eed8542208389efa60ec8454e06fd84b276b45b2e33897f9d7825")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-mbstrdecoder", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-typepy/package.py]: Deps are incorrect

**Review by adamjstewart — CHANGES_REQUESTED** [review]
If you're new to Python packaging it's helpful to read https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html

**Comment by francislz** [comment]
@adamjstewart Thanks for the review ! I changed the PR to make a simpler contribution and get myself used with Python packaging first. Sorry for the trouble.

File: var/spack/repos/builtin/packages/py-obspy/package.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+    homepage = "https://pypi.org/project/obspy/"
+    pypi = "obspy/obspy-1.4.0.tar.gz"
+
+    maintainers("francislz")
+
+    version("1.4.0", sha256="336a6e1d9a485732b08173cb5dc1dd720a8e53f3b54c180a62bb8ceaa5fe5c06")
+    version("1.2.1", sha256="340905ecc6d6d29dfc7103808e74e23e1fcfcbbc6ac32d07accce9a8b0322f47")
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-obspy/package.py:20]: Requires Python 3.8+

Missing deps:
```python
INSTALL_REQUIRES = [                                                                     
    'numpy>=1.20',                                                                       
    'scipy>=1.7',                                                                        
    'matplotlib>=3.3',                                                                   
    'lxml',                                                                              
    'setuptools',                                                                        
    'sqlalchemy',                                                                        
    'decorator',                                                                         
    'requests',                                                                          
]      
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by alecbcs** [comment]
Going to close this PR due to a lack of response after review feedback. @francislz  happy to reopen if you still want to get this merged and have time to work on it.

**Pull request closed without merging**